cylc/uiserver/resolvers.py

```diff
@@ -289,79 +269,75 @@ async def scan(
-
-            # build the command
-            cmd = ['cylc', 'play', '--color=never']
-            cmd = _build_cmd(cmd, args)
-
-        except Exception as exc:
-            # oh noes, something went wrong, send back confirmation
-            return cls._error(exc)
-        # start each requested flow
+        cylc_version = args.pop('cylc_version', None)
+        results: Dict[str, str] = {}
+        failed = False
         for tokens in workflows:
             try:
+                cmd = _build_cmd(['cylc', 'play', '--color=never'], args)
+
                 if tokens['user'] and tokens['user'] != getuser():
                     return cls._error(
                         'Cannot start workflows for other users.'
                     )
                 # Note: authorisation has already taken place.
                 # add the workflow to the command
-                cmd = [*cmd, tokens['workflow']]
+                wflow: str = tokens['workflow']
+                cmd = [*cmd, wflow]
 
                 # get a representation of the command being run
                 cmd_repr = ' '.join(cmd)
                 if cylc_version:
                     cmd_repr = f'CYLC_VERSION={cylc_version} {cmd_repr}'
                 log.info(f'$ {cmd_repr}')
 
-                # run cylc run
+                env = os.environ.copy()
+                env.pop('CYLC_ENV_NAME', None)
+                if cylc_version:
+                    env['CYLC_VERSION'] = cylc_version
+
+                # run cylc play
                 proc = Popen(
                     cmd,
+                    env=env,
                     stdin=DEVNULL,
                     stdout=PIPE,
                     stderr=PIPE,
                     text=True
                 )
-                ret = proc.wait(timeout=20)
+                ret_code = proc.wait(timeout=20)
 
-                if ret:
+                if ret_code:
                     # command failed
                     out, err = proc.communicate()
-                    msg = err.strip() or out.strip() or (
-                        f'Could not start {tokens["workflow"]}'
-                        f' - {cmd_repr}'
-                    )
-                    raise Exception(
-                        msg
+                    results[wflow] = err.strip() or out.strip() or (
+                        f'Command failed ({ret_code}): {cmd_repr}'
                     )
+                    failed = True
+                else:
+                    results[wflow] = 'started'
 
             except Exception as exc:
                 # oh noes, something went wrong, send back confirmation
                 return cls._error(exc)
 
-            else:
-                # send a success message
-                return cls._return(
-                    'Workflow started'
-                )
+        if failed:
+            return cls._error(
+                "\n".join(f"{wflow}: {msg}" for wflow, msg in results.items())
+            )
```

Comment:
```suggestion
        if failed:
            return cls._error(
                "\n\n" + "\n\n".join(
                    f"{wflow}: {msg}" for wflow, msg in results.items()
                )
            )
```

Results in:

![Screenshot from 2023-11-22 11-08-17](https://github.com/cylc/cylc-uiserver/assets/16705946/77a0bc4c-55e2-489d-aeca-dbc7913d7e54)

* The first `\n\n` is to separate the "Command failed: play -` message from the first result.
* The second `\n\n` is to put a blank line between each result as the stderr can be multiline.